include/sway/tree/view.h

```diff
@@ -286,15 +286,24 @@ void view_destroy(struct sway_view *view);
 void view_begin_destroy(struct sway_view *view);
 
 /**
- * Map a view, ie. make it visible in the tree.
+ * Map a view, ie. make it visible in the tree. view_premap must be called
+ * before calling view_map.
+ */
+void view_map(struct sway_view *view, struct wlr_surface *wlr_surface);
+
+/**
+ * Prepare the view for its upcoming mapping, sending the intended dimensions
+ * so that the first frame has a chance of being correct. If CSD preferences or
+ * floating tendency changes, this may turn out to be inaccurate but no worse
+ * than skipping the step.
  *
  * `fullscreen` should be set to true (and optionally `fullscreen_output`
  * should be populated) if the view should be made fullscreen immediately.
  *
  * `decoration` should be set to true if the client prefers CSD. The client's
  * preference may be ignored.
  */
-void view_map(struct sway_view *view, struct wlr_surface *wlr_surface,
+void view_premap(struct sway_view *view, struct wlr_surface *wlr_surface,
```

Comment:
I wonder if we can find a better name? Maybe "setup", "initial setup", "prepare"? (I'd be fine with "premap", just trying to see if there are better alternatives.)

Reply:
I named it `view_setup` for now, seems good enough.